net.solarnetwork.node.weather.ibm.wc/src/net/solarnetwork/node/weather/ibm/wc/BasicWCClient.java

```diff
@@ -77,14 +83,32 @@ private BigDecimal parseBigDecimal(JsonNode n) {
 		if ( n.isNull() ) {
 			return null;
 		}
-		return new BigDecimal(n.asLong());
+		return n.decimalValue();
 	}
 
 	private LocalTime parseTimeValue(JsonNode n) {
 		if ( n == null || n.isNull() ) {
 			return null;
 		}
-		return new LocalTime(n.asLong() * 1000);
+		try {
+			return new LocalTime(this.tsFormat.parse(n.asText()));
```

Comment:
Still not correct, I don't think, in that the output time will be in the JVM time zone, not the data's time zone. See my other comments on using Joda's `DateTimeFormatter` instead of `SimpleDateFormat` here.